cmake/Modules/FindSSE.cmake

```diff
@@ -0,0 +1,83 @@
+# check for SSE extensions
+include(CheckCXXSourceRuns)
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+    set(SSE_FLAGS)
+
+    set(CMAKE_REQUIRED_FLAGS "-msse3")
+    check_cxx_source_runs("
+        #include <pmmintrin.h>
+
+        int main()
+        {
+           __m128d a, b;
+           double vals[2] = {0};
+           a = _mm_loadu_pd(vals);
+           b = _mm_hadd_pd(a,a);
+           _mm_storeu_pd(vals, b);
+           return 0;
+        }"
+        HAS_SSE3_EXTENSIONS)
+
+    set(CMAKE_REQUIRED_FLAGS "-msse2")
+    check_cxx_source_runs("
+        #include <emmintrin.h>
+
+        int main()
+        {
+            __m128d a, b;
+            double vals[2] = {0};
+            a = _mm_loadu_pd(vals);
+            b = _mm_add_pd(a,a);
+            _mm_storeu_pd(vals,b);
+            return 0;
+         }"
+         HAS_SSE2_EXTENSIONS)
+
+    set(CMAKE_REQUIRED_FLAGS "-msse")
+    check_cxx_source_runs("
+        #include <xmmintrin.h>
+        int main()
+        {
+            __m128 a, b;
+            float vals[4] = {0};
+            a = _mm_loadu_ps(vals);
+            b = a;
+            b = _mm_add_ps(a,b);
+            _mm_storeu_ps(vals,b);
+            return 0;
+        }"
+        HAS_SSE_EXTENSIONS)
+
+    set(CMAKE_REQUIRED_FLAGS)
+
+    if(HAS_SSE3_EXTENSIONS)
```

Comment:
This essentially adds SSE3 as a dependency if available. That sounds a bit unnecessary. I think there should be some flag/parameter to `find_package()` to only force some specific instruction set. Just because your compiler is able to output SSE3 code doesn't necessarily mean that you'll want SSE3 code for backwards compatibility.